packages/shared/src/components/search/SearchResults/SearchResultsLayout.tsx

```diff
@@ -0,0 +1,145 @@
+import { PageWidgets } from '../../utilities';
+import { StraightArrowIcon } from '../../icons';
+import {
+  providerToIconMap,
+  providerToLabelTextMap,
+  SearchProviderButton,
+} from '../SearchPanel';
+import { SearchProviderEnum, SearchSuggestion } from '../../../graphql/search';
+import { useSearchResultsLayout } from '../../../hooks/search/useSearchResultsLayout';
+import { LogEvent, Origin, TargetType } from '../../../lib/log';
+import { useLogContext } from '../../../contexts/LogContext';
+import { webappUrl } from '../../../lib/constants';
+import { SearchResultsTags } from './SearchResultsTags';
+import { SearchResultsSources } from './SearchResultsSources';
+import { useFeedLayout } from '../../../hooks';
+import { useSearchProviderSuggestions } from '../../../hooks/search';
+
+type SearchResultsLayoutProps = PropsWithChildren;
+
+const AICta = {
+  Icon: providerToIconMap[SearchProviderEnum.Chat],
+  Label: providerToLabelTextMap[SearchProviderEnum.Chat],
+};
+
+export const SearchResultsLayout = (
+  props: SearchResultsLayoutProps,
+): ReactElement => {
+  const { children } = props;
+
+  const { isListMode } = useFeedLayout();
+  const { isSearchResultsUpgrade } = useSearchResultsLayout();
+  const {
+    query: { q: query },
+    push,
+  } = useRouter();
+  const { logEvent } = useLogContext();
+
+  const { isLoading: isTagsLoading, suggestions: suggestedTags } =
+    useSearchProviderSuggestions({
+      query: `${query}`,
+      provider: SearchProviderEnum.Tags,
+      limit: 10,
+    });
+  const tags = suggestedTags?.hits?.map(({ title }) => title) ?? [];
+
+  const { isLoading: isSourcesLoading, suggestions: suggestedSources } =
+    useSearchProviderSuggestions({
+      query: `${query}`,
+      provider: SearchProviderEnum.Sources,
+      limit: 10,
+    });
+  const sources = suggestedSources?.hits ?? [];
+
+  const onTagClick = (suggestion: SearchSuggestion) => {
+    const tag = suggestion.id || suggestion.title.toLowerCase();
+
+    logEvent({
+      event_name: LogEvent.Click,
+      target_type: TargetType.SearchRecommendation,
+      target_id: tag,
+      feed_item_title: tag,
+      extra: JSON.stringify({
+        origin: Origin.SearchPage,
+        provider: SearchProviderEnum.Tags,
+      }),
+    });
+
+    push(`${webappUrl}tags/${tag}`);
+  };
+
+  if (!isSearchResultsUpgrade) {
+    return <>{children}</>;
+  }
+
+  return (
+    <section className="mx-auto w-full laptopL:max-w-screen-laptop">
+      <div className="flex flex-row border-border-subtlest-tertiary laptop:-mx-16 laptop:pb-0 laptopL:mx-auto laptopL:border-x">
+        <div className="flex-1 border-r border-border-subtlest-tertiary">
+          <h2 className="px-4 py-4 font-bold text-text-primary typo-body">
+            Related posts
+          </h2>
+          <div
+            role="list"
+            className={classNames(
+              'mt-2.5',
+              isListMode
+                ? 'flex flex-col [&>article]:rounded-none'
```

Comment:
Why is this `[&>article]:rounded-none` needed?
The actual list vs card should have no special needs right?

Reply:
Because [on figma](https://www.figma.com/design/nmfWPS7x3kzLUvYMkBx2kW/daily.dev---Dev-Mode?node-id=13419-45716&t=I5YN0t9vaVkyaSw4-0) they are not rounded, should I skip on this detail?